src/test/storage.zig

```diff
@@ -215,6 +224,13 @@ pub const Storage = struct {
     ) void {
         storage.assert_bounds_and_alignment(buffer, offset);
 
+        // Verify that there are no concurrent overlapping writes.
```

Comment:
Nice check!

Reply:
Did this catch anything?

Reply:
It didn't, but I had the idea while looking at the WAL's sector locking logic.